lib/r10k/module_loader/stub_puppetfile.rb

```diff
@@ -0,0 +1,36 @@
+require 'r10k/module_loader/puppetfile'
+require 'r10k/module/stub'
+
+module R10K
+  module ModuleLoader
+
+    # This sub classes the Puppetfile and provides the same functionality
+    # except none of the modules instantiated are sync-able. They are only
+    # used to compare versions to decide if the real modules should be synced.
+    class StubPuppetfile < R10K::ModuleLoader::Puppetfile
```

Comment:
Also don't like Stub here. And since we don't need any of the purge info maybe it'd be better to factor out the parsing functionality and include it both here and in the normal puppetfile module loader rather than subclassing?

Reply:
I think I kind of like that idea, if it's not too difficult. Seems a little weird to have a subclass that is supposed to do _less_ than the parent.